### ISSUE TITLE
Incorrect parameter count with generic typeclasses and inheritance

### ISSUE THREAD

**Issue by ahah-figure** [PyCQA/pylint#10788]
### Bug description

I have something like the following in python 3.12.

```python
from dataclasses import dataclass


@dataclass
class One[T]:
    one: T


@dataclass
class Two[T](One[T]):
    two: T

one = One(1)
two = Two(1, 2)  # pylint says too-many-function-args
print(one, two)
```

pylint doesn't seem to recognize that the `Two` class requires two parameters at init. This only happens when I use TypeVars.

### Configuration

```ini

```

### Command used

```shell
pylint file.py
```

### Pylint output

```python
pl.py:14:6: E1121: Too many positional arguments for constructor call (too-many-function-args)
```

### Expected behavior

I would not expect any warning.

### Pylint version

```shell
pylint 4.0.4
astroid 4.0.2
Python 3.12.12 | packaged by conda-forge | (main, Oct 22 2025, 23:25:55) [GCC 14.3.0]
```

### OS / Environment

Ubuntu 22, bash

### Additional dependencies

```python

```
